hdf/src/hblocks.c

```diff
@@ -725,10 +721,8 @@ HLIstaccess(accrec_t *access_rec, int16 acc_mode)
     ret_value = HAregister_atom(AIDGROUP, access_rec);
 
 done:
-    if (ret_value == FAIL) { /* Error condition cleanup */
-        if (access_rec->special_info != NULL)
-            HDfree(access_rec->special_info);
-    }
+    if (ret_value == FAIL)
+        free(access_rec->special_info);
```

Comment:
Just standalone view - should "access_rec->special_info" be set to NULL? Not knowing the possible other use for the pointer, might be a good idea to future proof an invalid usage.

Reply:
See the function: HLPcloseAID

Reply:
I think there's a lot of instances where we should be resetting pointers to NULL after free.